src/dlgprefautodjdlg.ui

```diff
@@ -6,101 +6,104 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>610</width>
-    <height>173</height>
+    <width>637</width>
+    <height>218</height>
    </rect>
   </property>
   <property name="windowTitle">
    <string>AutoDJ Preferences</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
-   <item row="4" column="0">
+   <item row="0" column="0">
+    <widget class="QLabel" name="textAutoDjRequeue">
+     <property name="text">
+      <string>Re-queue tracks after playback</string>
+     </property>
+     <property name="buddy">
+      <cstring>ComboBoxAutoDjRequeue</cstring>
+     </property>
+    </widget>
+   </item>
+   <item row="0" column="2">
+    <widget class="QComboBox" name="ComboBoxAutoDjRequeue">
+     <property name="toolTip">
+      <string>Add a track to the end of the Auto DJ playlist once it is played, instead of removing it.</string>
+     </property>
+    </widget>
+   </item>
+   <item row="1" column="0" colspan="2">
+    <widget class="QLabel" name="autoDjMinimumAvailableLabel">
+     <property name="text">
+      <string>Minimum available tracks in Track Source</string>
+     </property>
+    </widget>
+   </item>
+   <item row="1" column="2">
+    <widget class="QSpinBox" name="autoDjMinimumAvailableSpinBox">
+     <property name="toolTip">
+      <string>This percentage of tracks are always available for selecting, regardless of when they were last played.</string>
+     </property>
+     <property name="suffix">
+      <string>%</string>
+     </property>
+     <property name="maximum">
+      <number>100</number>
+     </property>
+     <property name="value">
+      <number>20</number>
+     </property>
+    </widget>
+   </item>
+   <item row="2" column="2">
+    <widget class="QCheckBox" name="autoDjIgnoreTimeCheckBox">
+     <property name="toolTip">
+      <string>Uncheck, to ignore all played tracks.</string>
+     </property>
+     <property name="text">
+      <string>Suspend track in Track Source from re-queue</string>
+     </property>
+    </widget>
+   </item>
+   <item row="3" column="0">
+    <widget class="QLabel" name="autoDjMinimumAvailableLabel_2">
+     <property name="text">
+      <string>Suspension period for track selection</string>
+     </property>
+    </widget>
+   </item>
+   <item row="3" column="2">
+    <widget class="QTimeEdit" name="autoDjIgnoreTimeEdit">
+     <property name="toolTip">
+      <string>Duration after which a track is eligible for selection by Auto DJ again</string>
+     </property>
+     <property name="displayFormat">
+      <string>hh:mm</string>
+     </property>
+     <property name="timeSpec">
+      <enum>Qt::LocalTime</enum>
+     </property>
+    </widget>
+   </item>
+   <item row="4" column="2">
+    <widget class="QCheckBox" name="autoDjIncludeLibTracksCheckBox">
+     <property name="text">
+      <string>Include library tracks for random addition </string>
+     </property>
+    </widget>
+   </item>
+   <item row="5" column="1" colspan="2">
```

Comment:
The expanding behavior has changed, before labels and controls where expanded. now only the label expands.